includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1372,4 +1258,295 @@ public function get_product_type( $product_id ) {
 			return false;
 		}
 	}
+
+	/**
+	 * Add ability to get products by 'reviews_allowed' in WC_Product_Query.
+	 *
+	 * @since 3.2.0
+	 * @param string $where where clause
+	 * @param WP_Query $wp_query
+	 */
+	public function reviews_allowed_query_where( $where, $wp_query ) {
```

Comment:
It's a filter. Filters have to be public otherwise you'll get something like this:
`
call_user_func_array() expects parameter 1 to be a valid callback, cannot access protected method WC_Product_Data_Store_CPT::reviews_allowed_query_where()`